python/ray/data/impl/block_builder.py

```diff
@@ -6,6 +6,30 @@
 class BlockBuilder(Generic[T]):
     """A builder class for blocks."""
 
+    @staticmethod
+    def for_block(block_type) -> "BlockBuilder[T]":
```

Comment:
I think you can rely on accessor instead o do:

`return BlockAccessor.for_block(block).builder()`

Reply:
Ok, good to know! It seems make more sense to have an API in BlockBuilder than BlockAccessor to create the builder instances. We may add this interface and/or make accessor to call builder's build(), or drop this change for now since it's not an actual issue. What do you think?

Reply:
Either seems fine to me!

Reply:
Will continue this PR then, BlockBuilder.for_block_type(...) seems more intuitive.